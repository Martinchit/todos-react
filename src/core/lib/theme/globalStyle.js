import { createGlobalStyle } from 'styled-components';

export const Theme = {
  fonts: {},
  colors: {
    primary: '#3772ff',
    secondary: '#F4F4F4',
    success: '#2bab24',
    danger: '#E71D36',
    warning: '#FF9F1C',
    black: '#011627',
    white: '#FDFFFC',
  },
};

export const GlobalStyle = createGlobalStyle`
  * {
    padding: 0;
    margin: 0;
    border: 0;
    font-family: 'Source Sans Pro', sans-serif;
    font-size: 16px;
  }

  body {
    min-height: 100vh;
    min-width: 100vw;
    background-color: ${Theme.colors.secondary};
  }
  
  h1, h2, h3, h4, h5, h6 {
    letter-spacing: normal;
  }
`;
