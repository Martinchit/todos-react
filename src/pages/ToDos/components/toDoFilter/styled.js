import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { IoCheckmarkDoneCircleSharp, IoAlertCircleSharp } from 'react-icons/io5';
import { Box } from '../../../../core/components/atoms/Box';

export const ToDoFilterContainer = styled(Box)`
  box-shadow: ${({ theme }) => `0 -5px 5px -7px ${theme.colors.black}`};
`;

export const ToDoStatsDoneIcon = styled(IoCheckmarkDoneCircleSharp)`
  color: ${({ theme }) => theme.colors.success};
  font-size: 1.5rem;
  margin-right: 3px;
`;

export const ToDoStatsActiveIcon = styled(IoAlertCircleSharp)`
  color: ${({ theme }) => theme.colors.warning};
  font-size: 1.5rem;
  margin-right: 3px;
`;

export const ToDoFilterOption = styled(Link)`
  margin: 0 3px;
  border-bottom: ${({ $selected, theme }) =>
    $selected ? `1px solid ${theme.colors.primary}` : 'none'};
  font-weight: ${({ $selected }) => ($selected ? 800 : 400)};
  color: ${({ theme }) => theme.colors.black};
  text-decoration: none;
  padding: 0 10px;
  cursor: pointer;
`;

export const CompletedToDoDeleteButton = styled.button`
  color: ${({ theme }) => theme.colors.danger};
  cursor: pointer;
  font-weight: 600;
`;
