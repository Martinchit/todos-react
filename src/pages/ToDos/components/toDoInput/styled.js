import styled from 'styled-components';
import { TiDeleteOutline, TiArrowSortedUp, TiArrowSortedDown } from 'react-icons/ti';

export const ToDoInputField = styled.input`
  width: 80%;
  padding: 2%;
  background-color: transparent;

  :focus,
  :active,
  :hover {
    outline: none;
  }

  ::placeholder,
  ::-webkit-input-placeholder {
    font-style: italic;
  }
  :-ms-input-placeholder {
    font-style: italic;
  }
`;

export const DeleteIcon = styled(TiDeleteOutline)`
  font-size: 1.5rem;
  margin: 0 3%;
  cursor: pointer;
  visibility: ${({ $visible }) => ($visible ? 'visible' : 'hidden')};
  color: ${({ theme }) => theme.colors.danger};
`;

export const ExpandIcon = styled(TiArrowSortedDown)`
  font-size: 1.5rem;
  margin: 0 3%;
  cursor: pointer;
  visibility: ${({ $visible }) => ($visible ? 'visible' : 'hidden')};
  color: ${({ theme }) => theme.colors.primary};
`;

export const HideIcon = styled(TiArrowSortedUp)`
  font-size: 1.5rem;
  margin: 0 3%;
  cursor: pointer;
  visibility: ${({ $visible }) => ($visible ? 'visible' : 'hidden')};
  color: ${({ theme }) => theme.colors.black};
`;
