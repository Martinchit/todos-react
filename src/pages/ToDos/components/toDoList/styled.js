import styled from 'styled-components';
import { RiDeleteBinLine } from 'react-icons/ri';
import { Box } from '../../../../core/components/atoms/Box';

export const ToDoItemContainer = styled(Box)`
  box-shadow: ${({ theme }) => `0 -5px 5px -7px ${theme.colors.black}`};
`;

export const ToDoLabel = styled.label`
  width: 80%;
  cursor: pointer;
`;

export const ToDoEditInput = styled.input`
  height: 100%;
  width: 86%;
  background-color: transparent;

  :focus,
  :active,
  :hover {
    outline: none;
  }

  :focus {
    border-bottom: ${({ theme }) => `2px solid ${theme.colors.success}`};
  }
`;

export const ItemDeleteButton = styled(RiDeleteBinLine)`
  font-size: 1.5rem;
  cursor: pointer;
  color: ${({ theme }) => theme.colors.danger};
`;
