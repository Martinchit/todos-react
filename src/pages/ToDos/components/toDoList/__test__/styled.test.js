import React from "react";
import { shallow } from 'enzyme'; 
import 'jest-styled-components';

import { ToDoItemContainer, ToDoLabel, ToDoEditInput, ItemDeleteButton } from "../styled";
import { Theme } from '../../../../../core/lib/theme/globalStyle';

describe("ToDoList styled components", () => {
  it("renders correctly", () => {
    const tDIC = shallow(<ToDoItemContainer theme={Theme}  />)
    expect(tDIC).toMatchSnapshot()

    const tDL = shallow(<ToDoLabel $visible={true} theme={Theme}  />)
    expect(tDL).toMatchSnapshot()

    const tDEI = shallow(<ToDoEditInput $visible={true} theme={Theme}  />)
    expect(tDEI).toMatchSnapshot()

    const iDB = shallow(<ItemDeleteButton $visible={true} theme={Theme}  />)
    expect(iDB).toMatchSnapshot()
  })
})