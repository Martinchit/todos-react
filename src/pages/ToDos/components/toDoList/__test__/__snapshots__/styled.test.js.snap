// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDoList styled components renders correctly 1`] = `ShallowWrapper {}`;

exports[`ToDoList styled components renders correctly 2`] = `ShallowWrapper {}`;

exports[`ToDoList styled components renders correctly 3`] = `ShallowWrapper {}`;

exports[`ToDoList styled components renders correctly 4`] = `ShallowWrapper {}`;
